npm.el

```diff
@@ -22,41 +22,47 @@
 
 ;;; Code:
 
-(require 'cl)
+(eval-when-compile
+  (require 'cl))
```

Comment:
I haven't looked closely, but what is CL actually being used for in here? I vaguely recall from looking at it that it might just be some simple list functions, and I wonder if those can be rewritten and nuke this dependency.